kolibri/auth/test/test_permissions_classes.py

```diff
@@ -46,23 +47,26 @@ def test_user_cannot_delete(self):
 class TestBooleanOperationsOnPermissionClassesTestCase(TestCase):
 
     def setUp(self):
+        self.facility = Facility.objects.create()
         self.obj = object()
-        self.user = FacilityUser.objects.create(username='dummyuser')
+        self.user = FacilityUser.objects.create(username='dummyuser', facility=self.facility)
         self.queryset = FacilityUser.objects.all()
 
-    def assertAllowAll(self, perms):
+    def assertAllowAll(self, perms, test_filtering=True):
         self.assertTrue(perms.user_can_create_object(self.user, self.obj))
         self.assertTrue(perms.user_can_read_object(self.user, self.obj))
         self.assertTrue(perms.user_can_update_object(self.user, self.obj))
         self.assertTrue(perms.user_can_delete_object(self.user, self.obj))
-        self.assertSetEqual(self.queryset, perms.readable_by_user_filter(self.user, self.queryset))
+        if test_filtering:
+            self.assertSetEqual(set(self.queryset), set(perms.readable_by_user_filter(self.user, self.queryset)))
```

Comment:
I've gotten into trouble with `set`s of model instances before. Objects in sets are identified by their hashes, so two objects with different hashes are considered distinct. Note that a Model instance's hash is it's pk.